Course/Course/Managers/CalendarManager.swift

```diff
@@ -0,0 +1,482 @@
+    var isOn: Bool
+    var modalPresented: Bool
+}
+
+class CalendarManager: NSObject {
+    
+    private let courseName: String
+    private let courseID: String
+    private let courseStructure: CourseStructure?
+    private let config: ConfigProtocol
+    
+    private let eventStore = EKEventStore()
+    private let iCloudCalendar = "icloud"
+    private let alertOffset = -1
+    private let calendarKey = "CalendarEntries"
+    
+    private var localCalendar: EKCalendar? {
+        if authorizationStatus != .authorized { return nil }
+        
+        var calendars = eventStore.calendars(for: .event).filter { $0.title == calendarName }
+        
+        if calendars.isEmpty {
+            return nil
+        } else {
+            let calendar = calendars.removeLast()
+            // calendars.removeLast() pop the element from array and after that,
+            // following is run on remaing members of array to remove them
+            // calendar app, if they had been added.
+            calendars.forEach { try? eventStore.removeCalendar($0, commit: true) }
+            
+            return calendar
+        }
+    }
+    
+    private let calendarColor = Theme.Colors.accentColor
+    
+    private var calendarSource: EKSource? {
+        eventStore.refreshSourcesIfNecessary()
+        
+        let iCloud = eventStore.sources.first(
+            where: { $0.sourceType == .calDAV && $0.title.localizedCaseInsensitiveContains(iCloudCalendar) })
+        let local = eventStore.sources.first(where: { $0.sourceType == .local })
+        let fallback = eventStore.defaultCalendarForNewEvents?.source
+        
+        return iCloud ?? local ?? fallback
+    }
+        
+    private func calendar() -> EKCalendar {
+        let calendar = EKCalendar(for: .event, eventStore: eventStore)
+        calendar.title = calendarName
+        calendar.cgColor = calendarColor.cgColor
+        calendar.source = calendarSource
+        
+        return calendar
+    }
+    
+    var authorizationStatus: EKAuthorizationStatus {
+        return EKEventStore.authorizationStatus(for: .event)
+    }
+    
+    var calendarName: String {
+        return config.platformName + " - " + courseName
+    }
+
+    private lazy var branchEnabled: Bool = {
+        return config.branch.enabled
+    }()
+    
+    var syncOn: Bool {
+        get {
+            if let calendarEntry = calendarEntry,
+               let localCalendar = localCalendar {
+                if calendarEntry.identifier == localCalendar.calendarIdentifier {
+                    return calendarEntry.isOn
+                }
+            } else {
+                if let localCalendar = localCalendar {
+                    let courseCalendar = CourseCalendar(
+                        identifier: localCalendar.calendarIdentifier,
+                        courseID: courseID,
+                        title: calendarName,
+                        isOn: true,
+                        modalPresented: false)
```

Comment:
Could you move the bracket to the next line to respect the "parameter on its own line" style?